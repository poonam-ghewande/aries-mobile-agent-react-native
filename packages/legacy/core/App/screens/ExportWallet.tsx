import { useNavigation } from '@react-navigation/core'
import React, { useState } from 'react'
import { View, Text, StyleSheet, TouchableOpacity, ScrollView, Dimensions } from 'react-native'

import Button, { ButtonType } from '../components/buttons/Button'
import { useTheme } from '../contexts/theme'
import { Screens } from '../types/navigators'

interface PhraseData {
  id: number
  value: string
}

const Passphrase: React.FC = () => {
  const { ColorPallet, TextTheme } = useTheme()
  const navigation = useNavigation()
  const [phraseData, setPhraseData] = useState<PhraseData[]>([
    { id: 1, value: 'abandon' },
    { id: 2, value: 'ability' },
    { id: 3, value: 'able' },
    { id: 4, value: 'about' },
    { id: 5, value: 'above' },
    { id: 6, value: 'absurd' },
    { id: 7, value: 'abuse' },
    { id: 8, value: 'absorb' },
  ])
  const screen = Dimensions.get('screen')

  const styles = StyleSheet.create({
    container: {
      justifyContent: 'center',
      height: screen.height,
      flex: 1,
    },
    subcontainer: {
      flexDirection: 'row',
      flexWrap: 'wrap',
      justifyContent: 'center',
      alignSelf: 'center',
      marginTop: '30%',
    },
    rowContainerInternalView: {
      flexDirection: 'row',
      width: 150,
      margin: 10,
      padding: 7,
      justifyContent: 'center',
      alignItems: 'center',
      borderRadius: 5,
      borderWidth: 1.3,
      backgroundColor: ColorPallet.brand.primaryBackground,
    },
    rowContainerView: {
      flexDirection: 'row',
      alignItems: 'center',
      marginRight: 20,
    },
    mediumTextStyle: {
      fontSize: 18,
      backgroundColor: ColorPallet.brand.primaryBackground,
    },
    textStyle: {
      width: 20,
      textAlign: 'center',
    },
    scrollView: {
      flex: 1,
    },
    containerView: {
      flex: 1,
      alignItems: 'center',
      paddingTop: 10,
    },
    titleText: {
      fontSize: 26,
      marginTop: 15,
      textAlign: 'center',
    },
    detailText: {
      fontSize: 18,
      marginHorizontal: 30,
      marginTop: 20,
      lineHeight: 20,
      textAlign: 'center',
    },
  })

  return (
    <ScrollView contentContainerStyle={styles.container}>
      <View>
        <View>
          <Text style={[TextTheme.title, styles.titleText]}>Write down your seed phrase</Text>
          <Text style={[TextTheme.label, styles.detailText]}>
            This is your seed phrase. Write it down on a paper and keep it in a safe place. You'll be asked to re-enter
            this phrase (in order) on the next step.
          </Text>
        </View>
      </View>
      <View style={styles.subcontainer}>
        {phraseData.map((item: PhraseData) => (
          <TouchableOpacity key={item.id}>
            <View style={styles.rowContainerView}>
              <View
                style={[
                  styles.rowContainerInternalView,
                  {
                    borderStyle: 'dashed',
                    borderColor: ColorPallet.brand.primary,
                  },
                ]}
              >
                <Text style={[TextTheme.labelSubtitle, styles.mediumTextStyle]}>{item.value}</Text>
              </View>
            </View>
          </TouchableOpacity>
        ))}
      </View>
      <View style={{ marginTop: 'auto', margin: 20, alignItems: 'stretch' }}>
        <Button
          title={'Continue'}
          accessibilityLabel={'Okay'}
          buttonType={ButtonType.Primary}
          onPress={() => navigation.navigate(Screens.PassphraseVerify as never)}
        />
      </View>
    </ScrollView>
  )
}

export default Passphrase
