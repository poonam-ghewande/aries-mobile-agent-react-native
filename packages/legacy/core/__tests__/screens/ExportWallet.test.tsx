import { render } from '@testing-library/react-native'
import React from 'react'

import ExportWallet from '../../App/screens/ExportWallet'
import renderer from 'react-test-renderer'
// import { testIdWithKey } from '../../App/utils/testable'

jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})
describe('Export wallet screen', () => {
  beforeEach(() => {
    // Silence console.error because it will print a warning about Switch
    // "Warning: dispatchCommand was called with a ref ...".
    jest.spyOn(console, 'error').mockImplementation(() => {
      return null
    })
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  test('screen renders correctly', () => {
    const tree = render(<ExportWallet />).toJSON()
    console.log('tree', tree)
    expect(tree).toMatchSnapshot()
  })

  // const navigation = {
  //   navigate: jest.fn(),
  //   goBack: jest.fn(),
  //   // Add other methods that your component uses if necessary
  // }

  // Your test case

  //   test('pressables exist', async () => {
  //     const { findByTestId } = render(<Developer />)

  //     const VerifierToggle = await findByTestId(testIdWithKey('ToggleVerifierCapability'))
  //     const ConnectionToggle = await findByTestId(testIdWithKey('ToggleConnectionInviterCapabilitySwitch'))
  //     const DevVerifierToggle = await findByTestId(testIdWithKey('ToggleDevVerifierTemplatesSwitch'))

  //     expect(VerifierToggle).not.toBe(null)
  //     expect(ConnectionToggle).not.toBe(null)
  //     expect(DevVerifierToggle).not.toBe(null)
  //   })
})
