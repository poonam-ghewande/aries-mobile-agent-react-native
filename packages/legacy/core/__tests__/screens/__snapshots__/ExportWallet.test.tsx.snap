// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Export wallet screen screen renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flex": 1,
      "height": 1334,
      "justifyContent": "center",
    }
  }
>
  <View>
    <View>
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 20,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 26,
                "marginTop": 15,
                "textAlign": "center",
              },
            ]
          }
        >
          Backup.write_down
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 14,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 18,
                "lineHeight": 20,
                "marginHorizontal": 30,
                "marginTop": 20,
                "textAlign": "center",
              },
            ]
          }
        >
          Backup.this_is_your_seed
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "marginTop": "30%",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginRight": 20,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#000000",
                  "borderRadius": 5,
                  "borderWidth": 1.3,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "margin": 10,
                  "padding": 7,
                  "width": 150,
                },
                Object {
                  "borderColor": "#42803E",
                  "borderStyle": "dashed",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "fontWeight": "normal",
                  },
                  Object {
                    "backgroundColor": "#000000",
                    "fontSize": 18,
                  },
                ]
              }
            >
              abandon
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginRight": 20,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#000000",
                  "borderRadius": 5,
                  "borderWidth": 1.3,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "margin": 10,
                  "padding": 7,
                  "width": 150,
                },
                Object {
                  "borderColor": "#42803E",
                  "borderStyle": "dashed",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "fontWeight": "normal",
                  },
                  Object {
                    "backgroundColor": "#000000",
                    "fontSize": 18,
                  },
                ]
              }
            >
              ability
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginRight": 20,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#000000",
                  "borderRadius": 5,
                  "borderWidth": 1.3,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "margin": 10,
                  "padding": 7,
                  "width": 150,
                },
                Object {
                  "borderColor": "#42803E",
                  "borderStyle": "dashed",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "fontWeight": "normal",
                  },
                  Object {
                    "backgroundColor": "#000000",
                    "fontSize": 18,
                  },
                ]
              }
            >
              able
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginRight": 20,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#000000",
                  "borderRadius": 5,
                  "borderWidth": 1.3,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "margin": 10,
                  "padding": 7,
                  "width": 150,
                },
                Object {
                  "borderColor": "#42803E",
                  "borderStyle": "dashed",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "fontWeight": "normal",
                  },
                  Object {
                    "backgroundColor": "#000000",
                    "fontSize": 18,
                  },
                ]
              }
            >
              about
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginRight": 20,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#000000",
                  "borderRadius": 5,
                  "borderWidth": 1.3,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "margin": 10,
                  "padding": 7,
                  "width": 150,
                },
                Object {
                  "borderColor": "#42803E",
                  "borderStyle": "dashed",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "fontWeight": "normal",
                  },
                  Object {
                    "backgroundColor": "#000000",
                    "fontSize": 18,
                  },
                ]
              }
            >
              above
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginRight": 20,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#000000",
                  "borderRadius": 5,
                  "borderWidth": 1.3,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "margin": 10,
                  "padding": 7,
                  "width": 150,
                },
                Object {
                  "borderColor": "#42803E",
                  "borderStyle": "dashed",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "fontWeight": "normal",
                  },
                  Object {
                    "backgroundColor": "#000000",
                    "fontSize": 18,
                  },
                ]
              }
            >
              absurd
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginRight": 20,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#000000",
                  "borderRadius": 5,
                  "borderWidth": 1.3,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "margin": 10,
                  "padding": 7,
                  "width": 150,
                },
                Object {
                  "borderColor": "#42803E",
                  "borderStyle": "dashed",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "fontWeight": "normal",
                  },
                  Object {
                    "backgroundColor": "#000000",
                    "fontSize": 18,
                  },
                ]
              }
            >
              abuse
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginRight": 20,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#000000",
                  "borderRadius": 5,
                  "borderWidth": 1.3,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "margin": 10,
                  "padding": 7,
                  "width": 150,
                },
                Object {
                  "borderColor": "#42803E",
                  "borderStyle": "dashed",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "fontWeight": "normal",
                  },
                  Object {
                    "backgroundColor": "#000000",
                    "fontSize": 18,
                  },
                ]
              }
            >
              absorb
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "margin": 20,
          "marginTop": "auto",
        }
      }
    >
      <View
        accessibilityLabel="Okay"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#42803E",
            "borderRadius": 4,
            "opacity": 1,
            "padding": 16,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                false,
                false,
                false,
              ]
            }
          >
            Continue
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
